sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala

```diff
@@ -67,15 +67,14 @@ sealed abstract class SValue {
         case SBool(x) => V.ValueBool(x)
         case SUnit => V.ValueUnit
         case SDate(x) => V.ValueDate(x)
-        case r: SRecord =>
-          V.ValueRecord(
-            maybeEraseTypeInfo(r.id),
-            (r.fields.toSeq.view zip r.values.iterator().asScala)
-              .map { case (field, sv) =>
-                (maybeEraseTypeInfo(field), go(sv, nextMaxNesting))
-              }
-              .to(ImmArray),
-          )
+        case SRecord(id, names0, values0) =>
+          val n = values0.asScala.reverseIterator.dropWhile(_ == SValue.SValue.None).size
```

Comment:
Surely, when we take the `n` members of `names0`, we don't know if they map to `None`s or `Some`s?

Reply:
The names and values are lists that map directly to eachother, you could probably unzip them to a list of tuples directly.

so say you have `names0 = ["first", "second", "third"]` and `values0 = [something, something, None]`
you'll get an `n` of 2, then end up with just `first` and `second` both as `something`

Reply:
If I understand things correctly, we're saying that all the `None` values are at the end of the `values0` list?

On that basis, I think we need a comment here so that assumption is made clear.

Reply:
Oh right, the logic is to _only_ drop Nones at the end, thus the `reverseIterator.dropWhile`, which is different to a filter.
if we had `names0 = ["first", "second", "third"]` and `values0 = [None, something, None]`
we'd end up only dropping `third`

Reply:
Its not an assumption, but rather a property of the normalisation

Reply:
Oh, that definitely needs a comment IMHO. That wasn't clear to myself at all.

Reply:
@remyhaemmerle-da Perhaps just a comment on the import logic that says "Trailing None fields are dropped"?